pkg/endpoint/policy.go

```diff
@@ -363,7 +363,9 @@ func (e *Endpoint) updateRealizedState(stats *regenerationStatistics, origDir st
 		e.syncPolicyMapController()
 	}
 
-	e.realizedBPFConfig = e.desiredBPFConfig
+	// Regeneration was successful - we can remove any state from prior versions
+	// of Cilium that needs to be cleaned up.
+	e.RemoveOldMaps()
```

Comment:
+ "Remove after v1.7"

Good catch, I didn't think of this upgrade case.

Reply:
Actually, why can't this be in the map sweeper? AFAICT that's only run after `<-restoreComplete` in the daemon so it should be safe to remove all `ep_config_*` at that point..?

Reply:
You're right, removed the need for the explicit function.